src/EditorFeatures/TestUtilities/CodeActions/CSharpCodeFixVerifier`2.cs

```diff
@@ -72,5 +74,48 @@ public static async Task VerifyCodeFixAsync(string source, DiagnosticResult[] ex
             test.ExpectedDiagnostics.AddRange(expected);
             await test.RunAsync();
         }
+
+        public static async Task VerifyCodeFixAsync((string first, string second) sources, (string first, string second) fixedSources, int numberOfFixAllIterations)
+            => await VerifyCodeFixAsync(sources, fixedSources, DiagnosticResult.EmptyDiagnosticResults, numberOfFixAllIterations);
+
+        public static async Task VerifyCodeFixAsync((string first, string second) sources, (string first, string second) fixedSources, DiagnosticResult[] expected, int numberOfFixAllIterations)
+        {
+            var test = new Test
+            {
+                TestState = { Sources = { sources.first, sources.second } },
+                FixedState = { Sources = { fixedSources.first, fixedSources.second } },
+                NumberOfFixAllInDocumentIterations = numberOfFixAllIterations
+            };
+
+            test.ExpectedDiagnostics.AddRange(expected);
+            await test.RunAsync();
+        }
+
+        /// <summary>
+        /// Verifies fix for a single diagnostic and iterative batch fix for all diagnostics.
+        /// </summary>
+        /// <param name="source">Original source with multiple diagnostics.</param>
+        /// <param name="fixedSource">Fixed source with code fix applied for a single diagnostic selected by <paramref name="diagnosticSelector"/>.</param>
+        /// <param name="batchFixedSource">Fixed source for iterative batch fix for all diagnostics.</param>
+        /// <param name="diagnosticSelector">Delegate to select a single diagnostic to apply fix to verify <paramref name="fixedSource"/>.</param>
+        public static async Task VerifyFixOneAndFixBatchAsync(
```

Comment:
📝 So far this is really an edge case. I'd like to continue without the overload for a bit to see how things are shaping up.

Reply:
I don't know if it is really an edge case to have tests which have multiple diagnostics and want to validate a fix on only one of them. I found it very hard to read or write the code that is within this method. In fact, I don't even know how to write it in VB as the only approach here seems to be highly nested object initializers - I don't know if VB also supports it or not. Can you paste a VB snippet that can accomplish the same functionality as this helper?